src/main/java/de/uni_stuttgart/it_rex/course/web/rest/written/ProgressResource.java

```diff
@@ -0,0 +1,219 @@
+        this.courseService = newCourseService;
+    }
+
+    /**
+     * {@code GET  /courses/:courseId} : get the progress for course "courseId" for the requesting user.
+     *
+     * @param courseId the id of the course to retrieve progress for.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)}
+     * and with body the chapter, or with status {@code 404 (Not Found)}.
+     */
+    @GetMapping("/courses/{courseId}")
+    public ResponseEntity<CourseProgressTrackerDTO> getCourseProgress(
+        @PathVariable final UUID courseId) {
+        log.debug("REST request to get Course Progress for Course : {}",
+            courseId);
+
+        // check that a course with the given id exists
+        // might want to compare against the list of the user's courses
+        // instead of asking the courseservice like this
+        if (courseService.findOne(courseId).isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+
+        UUID userId = RexAuthz.getUserId();
+        CourseProgressTrackerDTO progress =
+            progressTrackingService.findOrCreateCourseProgressTracker(courseId, userId);
+        return ResponseUtil.wrapOrNotFound(Optional.of(progress));
+    }
+
+    /**
+     * {@code PUT  /courses/{courseTrackerId}/contentReference : set the last accessed content reference for a specific user}
+     *
+     * @param courseTrackerId the user's tracker to update
+     * @param contentReferenceDTO the contentReference to store as last accessed
+     * @return {@link ResponseEntity} with status {@code 200 (OK)} and the updated
+     * {@link CourseProgressTrackerDTO} in its body, or with {@code 404 (Not Found)}
+     * if there is not tracker with the given Id or the contentReferenceDTO was invalid.
+     */
+    @PutMapping("/courses/{courseTrackerId}/contentReference")
+    public ResponseEntity<CourseProgressTrackerDTO> updateLastAccessedContentReference(
+        @PathVariable final UUID courseTrackerId,
+        @RequestBody final ContentReferenceDTO contentReferenceDTO){
+        CourseProgressTrackerDTO result = progressTrackingService.updateLastAccessedContentReference(courseTrackerId,contentReferenceDTO);
+
+        return ResponseEntity.ok().
+            headers(HeaderUtil.createEntityUpdateAlert(
+                applicationName,true, ENTITY_NAME_COURSEPT, result.getId().toString())).
+            body(result);
+    }
+
+    /**
+     * {@code GET  /content/:trackerId} : Get Content Progress Tracker for Content Item for User
+     *
+     * @param trackerId id of Content Progress Tracker
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)}
+     * and with the {@link ContentProgressTrackerDTO} in the body,
+     * or with status {@code 404 (Not Found)} in case of an invalid id.
+     */
+    @GetMapping("/content/{trackerId}")
+    public ResponseEntity<ContentProgressTrackerDTO> getContentProgress(
```

Comment:
Wird glaube ich im Frontend nicht benutzt und ist auch ein wenig unsinnig, wenn du die TrackerID brauchst, weil du die ja nur hast, wenn du den Progress schon hast (z.B. aus dem courseProgressDTO, wo die ganzen Tracker in dem Array mit drinstehen).

Glaube das Frontend benutzt tatsächlich auch nur den getCourseProgress-Endpunkt, unter anderem aus diesem Grund.

Löschen würde ich das ganze aber trotzdem nichts, weil kostet nichts und macht nichts kaputt.